docs/Toy/toy/interpreter.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import annotations
+
+import operator
+from typing import Any
+from itertools import accumulate
+
+from dataclasses import dataclass
+
+from xdsl.dialects.builtin import TensorType, VectorType, ModuleOp
+from xdsl.interpreter import Interpreter, InterpreterFunctions, register_impls, impl
+from xdsl.utils.exceptions import InterpretationError
+
+from . import dialect as toy
+
+
+@dataclass
+class Tensor:
```

Comment:
Btw how do we/do we want to generalize this to more types at some point? Will all of them just have this "reimplementation class" for the interpreter that simulates their behavior?

Reply:
Yep. I think one fun part will come when we interpret multiple levels, like when a pointwise tensor addition lowers to a buffer addition, represented by different classes in Python. I imagine a cast operation that is a no-op in the IR, but in the interpreter would need to dispatch to a function to convert from one to the other. Let's cross that bridge when we get to it